validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java

```diff
@@ -49,6 +50,12 @@ public interface ValidatorConfiguration extends ConversionServiceProvider {
     @NonNull
     ConstraintValidatorRegistry getConstraintValidatorRegistry();
 
+    /**
+     * @return The constraint factory to use.
+     */
+    @NonNull
+    InternalConstraintValidatorFactory getConstraintValidatorFactory();
```

Comment:
It is weird we expose a method returning a @Internal Class in a public API?

Reply:
can you annotate the method with `@Internal` and/or preferably make it non-public?

Reply:
I can expose ConstraintValidatorFactory and fail if its not InternalConstraintValidatorFactory